modules/core/src/main/java/org/apache/synapse/message/processor/impl/ScheduledMessageProcessor.java

```diff
@@ -549,4 +549,10 @@ private void setMessageProcessorState(ProcessorState state) {
 		registry.newNonEmptyResource(REG_PROCESSOR_BASE_PATH + getName(), false, "text/plain", state.toString(),
 				MP_STATE);
 	}
+
+    private void deleteMessageProcessorState() {
+        if (registry.getResourceProperties(REG_PROCESSOR_BASE_PATH + getName()) != null) {
+            registry.delete(REG_PROCESSOR_BASE_PATH + getName());
```

Comment:
Unfortunately the WSO2Registry and ESBRegistry implementations are different on how the non existing resource is being deleted. Also the  WSO2Registry.delete() call be called from different  other components